src/extensions/step_handlers/wim_handler/src/wim_step_handler.cpp

```diff
@@ -39,6 +41,76 @@ static std::string make_preferred(const char* path, const char* file = nullptr)
     return preferred;
 }
 
+/**
+ * @brief Verifies that the file path exists.
+ *
+ * @param payloadFilePath The path to the file.
+ * @return HRESULT S_OK when the file exists; an error hresult, otherwise.
+ */
+static HRESULT EnsurePayloadExists(const std::string& payloadFilePath)
+{
+    if (ADUCPAL_access(payloadFilePath.c_str(), F_OK) != 0)
```

Comment:
this is win32-only code. we shouldn't be calling anything in ADUCPAL.

I'd suggest this:

```C++
BOOL FileExists(const char* path)
{
   const DWORD dwAttrib = GetFileAttributes(path);

  return (dwAttrib != INVALID_FILE_ATTRIBUTES && !(dwAttrib & FILE_ATTRIBUTE_DIRECTORY));
}
```

Reply:
(with the generic name of "FileExists" or the like rather than "EnsurePayloadExists" as this method has nothing to do with "payload")